src/main/java/software/amazon/nio/spi/s3/util/StringUtils.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package software.amazon.nio.spi.s3.util;
+
+import software.amazon.nio.spi.s3.S3Path;
+
+/**
+ *
+ */
+public class StringUtils {
+    public static String join(String separator, String[] elements, int startAt) {
```

Comment:
Possibly you can do this with a stream and collector or reuse `String.join()` with the sub array? https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.CharSequence...-

Reply:
Looking at the JDK code I found there is a StringJoin that is supposed to do help in joining strings :) I would not complicate more the implementation as being a utility method I want to keep the implementation as most efficient as possible.